### DISCUSSION TITLE
Google Oauth2 Setup Question

### DISCUSSION THREAD

**Discussion by NilsChr** [pocketbase/pocketbase#525] [Q&A] [answered]
From the documentation: 

```
// user authentication via OAuth2
const userAuthData2 = await client.users.authViaOAuth2('google', 'CODE', 'VERIFIER', 'REDIRECT_URL');
```

What are these 2 values supposed to be: 'CODE', 'VERIFIER'  ?

**Comment by ganigeorgiev** [comment, marked as answer]
PocketBase integrates OAuth2 with the PKCE flow.

"CODE" is the auth code that you get from the auth redirect response. "VERIFIER" is the code verifier that you've sent to Google before authorizing. Full OAuth2 web example could be found in the "Auth via OAuth2" section in https://pocketbase.io/docs/manage-users/#auth-via-oauth2.

**Reply by NilsChr** [reply]
Thank you!
